### PR TITLE
User class original kj

### LINKED ISSUE #3: Users
A **User** holds on to all of a user’s data. As a user, I should be able to:

- [ ] add recipes to favoriteRecipes

- [ ] remove recipes to favoriteRecipes

- [ ] Decide to cook a recipe that week (add to my recipesToCook)

- [ ] Filter my favoriteRecipes or recipesToCook by type

- [ ] Search any of my saved recipes by name or ingredient

### PR DESCRIPTION
**Pull request by kathrynljackson**
# Kathryn Jackson

- [ ] checked functionality of the webpage
- [x] received help from a mentor/collaborator?
- [x] added teammates as reviewers

## Notes
* There was no sample test data for the `User` class. As stated in the [original rubric](), the tests should run according to created sample test data as opposed to the data provided. For this reason, this file is quite long after adding three users' worth of sample test data.
* While recipe test data was included and all recipe-related tests passed, the sample test data for the `Recipe` class was incorrect according to the construction of the `Recipe` class. New sample test data for the `Recipe` class was added.

## Issues Addressed
* Closes #3 

## Questions
* n/a

## Problems or Challenges
* FIXED: coding `searchSavedRecipes` to search based on recipe name AND ingredients.

## Screenshots (if applicable)
* n/a

### CODE REVIEW

File: test/user-test.js [hunk 1]
```diff
@@ -1,61 +1,4503 @@
...
 describe('User', function() {
   let user;
-  let userInfo;
-  let recipe;
+  let recipes;
 
   beforeEach(function() {
-    userInfo = data[0];
-    user = new User(userInfo)
+    let sampleUserData = [
```
Comment (GreyMatteOr) [inline test/user-test.js:10]: This is a TON of sample data. I think we can shorten this considerably: 
- I'm sure user 2 and 3 is lovely, but honestly user 1 is all we need. 
- Also, can we just be a broke college student? Why do they have 3000 foods. Have them live off of only pancakes, and their pantry reflect that.

File: src/user.js [hunk 2]
```diff
@@ -18,11 +20,26 @@ class User {
...
+  filterRecipesToCook(type) {
+    let filteredtoCook = this.recipesToCook.filter(recipe => {
+      return recipe.tags.includes(type);
+  })
+  return filteredtoCook;
+ }
+
+  searchSavedRecipes(keyword) {
+    let filteredSavedRecipes = this.favoriteRecipes.filter(recipe => {
+      return recipe.name.includes(keyword) || recipe.ingredients.some(ingredient => ingredient.name.includes(keyword));
```
Comment (GreyMatteOr) [inline src/user.js:40]: Very nice. Love the elegance and ease of reading!

**Review by GreyMatteOr — CHANGES_REQUESTED** [review]
Nice job! Honestly, if we could just trim down the sample data..... then that would be perfect

File: test/user-test.js (hunk 1 above)
Comment (kathrynljackson) [reply test/user-test.js:10]: Love it. I'll shorten the sample data down to one user.

**Review by krogowsk531 — COMMENTED** [review]
Everything looks great!  Good work!

File: test/user-test.js (hunk 1 above)
Comment (kathrynljackson) [reply test/user-test.js:10]: These changes were made and the tests still run.

**Pull request merged**